examples/IBLevelSet/ex0/SetFluidGasSolidViscosity.h

```diff
@@ -0,0 +1,129 @@
+// Filename: SetFluidGasSolidViscosity.h
+// Created on Dec 14, 2017 by Nishant Nangia
+
+/////////////////////// INCLUDE GUARD ////////////////////////////////////
+
+#ifndef included_SetFluidGasSolidViscosity
+#define included_SetFluidGasSolidViscosity
+
+///////////////////////////// INCLUDES ///////////////////////////////////
+
+#include <Variable.h>
+#include <ibamr/AdvDiffHierarchyIntegrator.h>
+#include <ibtk/ibtk_utilities.h>
+#include <tbox/Pointer.h>
+
+namespace IBTK
+{
+class HierarchyMathOps;
+}
+
+/*!
+ * Pre processing call back function to be hooked into IBAMR::VCINSStaggeredHierarchyIntegrator class.
+ *
+ * \param rho_idx a patch data index for the current density variable maintained by the integrator.
+ * \param ctx is the pointer to SetFluidGasSolidViscosity class object.
+ *
+ * \TODO: Let's move this out of the global namespace and use "snake case" for static function names.
+ */
+void callSetFluidGasSolidViscosityCallbackFunction(int mu_idx,
+                                                   SAMRAI::tbox::Pointer<SAMRAI::hier::Variable<NDIM> > mu_var,
+                                                   SAMRAI::tbox::Pointer<IBTK::HierarchyMathOps> hier_math_ops,
```

Comment:
@amneetb, is there a reason to prefer passing around the `HierarchyMathOps` object instead of the `PatchHierarchy`?

Reply:
The preference is if one needs to find rho(phi, grad(phi)) etc, HierachyMathOps will have canned functions to do these ops.

Reply:
I would prefer to use the `CartGridFunction` interface for setting grid data (and to improve that interface if needed).